src/main/java/org/broadinstitute/hellbender/engine/ReadShard.java

```diff
@@ -2,241 +2,106 @@
-        return divideIntervalIntoShards(interval, shardSize, shardSize, shardPadding, readsSource, dictionary);
-    }
+        @Override
+        public int getEnd() {
+            return paddedInterval.getEnd();
+        }
 
-    /**
-     * Divide an interval into ReadShards. Each shard will cover up to shardSize bases, include shardPadding
-     * bases of extra padding on either side, and begin shardStep bases after the previous shard.
-     *
-     * @param interval interval to shard; must be on the contig according to the provided dictionary
-     * @param shardSize desired shard size; intervals larger than this will be divided into shards of up to this size
-     * @param shardStep each shard will begin this many bases away from the previous shard
-     * @param shardPadding desired shard padding; each shard's interval will be padded on both sides by this number of bases (may be 0)
-     * @param readsSource data source for reads
-     * @param dictionary sequence dictionary for reads
-     * @return List of {@link ReadShard} objects spanning the interval
-     */
-    public static List<ReadShard> divideIntervalIntoShards( final SimpleInterval interval, final int shardSize, final int shardStep, final int shardPadding, final ReadsDataSource readsSource, final SAMSequenceDictionary dictionary ) {
-        Utils.nonNull(interval);
-        Utils.nonNull(readsSource);
-        Utils.nonNull(dictionary);
-        Utils.validateArg(shardSize >= 1, "shardSize must be >= 1");
-        Utils.validateArg(shardStep >= 1, "shardStep must be >= 1");
-        Utils.validateArg(shardPadding >= 0, "shardPadding must be >= 0");
+        public SimpleInterval getInterval() {
+            return interval;
+        }
 
-        Utils.validateArg(IntervalUtils.intervalIsOnDictionaryContig(interval, dictionary), () ->
-                "Interval " + interval + " not within the bounds of a contig in the provided dictionary");
+        public SimpleInterval getPaddedInterval() {
+            return paddedInterval;
+        }
 
-        final List<ReadShard> shards = new ArrayList<>();
-        int start = interval.getStart();
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
 
-        while ( start <= interval.getEnd() ) {
-            int end = Math.min(start + shardSize - 1, interval.getEnd());
+            final ShardBoundary key = (ShardBoundary) o;
 
-            final SimpleInterval nextShardInterval = new SimpleInterval(interval.getContig(), start, end);
-            final SimpleInterval nextShardIntervalPadded = nextShardInterval.expandWithinContig(shardPadding, dictionary);
-            shards.add(new ReadShard(nextShardInterval, nextShardIntervalPadded, readsSource));
+            if (!interval.equals(key.interval)) {
```

Comment:
Done, although I think it's unnecessary now since I've also added null checks for interval and padded interval.